apps/finance/app/src/components/Transfers.js

```diff
@@ -179,253 +174,110 @@ class Transfers extends React.PureComponent {
-                    <a role="button" onClick={this.handleResetFilters}>
-                      Reset filters
-                    </a>
-                  )}
-                </p>
-              </NoTransfers>
-            ) : (
-              <div>
-                <FixedTable
-                  header={
-                    above('medium') && (
-                      <TableRow>
-                        <DateHeader title="Date" />
-                        <SourceRecipientHeader title="Source / Recipient" />
-                        <ReferenceHeader title="Reference" />
-                        <AmountHeader title="Amount" align="right" />
-                        <TableHeader />
-                      </TableRow>
-                    )
-                  }
-                >
-                  {filteredTransfers
-                    .sort(({ date: dateLeft }, { date: dateRight }) =>
-                      // Sort by date descending
-                      compareDesc(dateLeft, dateRight)
-                    )
-                    .slice(0, displayedTransfers)
-                    .map(transfer => (
-                      <TransferRow
-                        key={transfer.transactionHash}
-                        token={tokenDetails[toChecksumAddress(transfer.token)]}
-                        transaction={transfer}
-                        smallViewMode={below('medium')}
-                      />
-                    ))}
-                </FixedTable>
-                {displayedTransfers < filteredTransfers.length && (
-                  <Footer>
-                    <Button mode="secondary" onClick={this.showMoreTransfers}>
-                      Show Older Transfers
-                    </Button>
-                  </Footer>
-                )}
-              </div>
+      <section>
+        <Header>
+          <Title compactMode={compactMode}>
+            <span>Transfers </span>
+            <span>
+              {compactMode && (
+                <ToggleFiltersButton
+                  title="Toggle Filters"
+                  onClick={this.handleToggleFiltersClick}
+                />
+              )}
+            </span>
+          </Title>
+          <TransfersFilters
+            dateRangeFilter={selectedDateRange}
+            onDateRangeChange={this.handleDateRangeChange}
+            tokenFilter={selectedToken}
+            onTokenChange={this.handleTokenChange}
+            transferTypeFilter={selectedTransferType}
+            onTransferTypeChange={this.handleTransferTypeChange}
+            compactMode={compactMode}
+            opened={filtersOpened}
+            symbols={symbols}
+            transferTypes={TRANSFER_TYPES_STRING}
+            downloadFileName={this.getCsvFilename()}
+            downloadUrl={this.encodeDataToCsv(filteredTransfers, tokenDetails)}
+          />
+        </Header>
+        {filteredTransfers.length === 0 ? (
+          <NoTransfers>
+            <p>
+              No transfers found.{' '}
```

Comment:
Could we also add the date period here? I think it's a bit confusing that we automatically apply the dates so a lot of users will just see "No transfers found." if they haven't made a transaction in the current month.

Reply:
What about making it clearer that filters are applied in general?

> No transfers found matching the current filters.
> [Clear current filters]

/cc @dizzypaty

Reply:
Yes, I agree with making it general – how about we say: 

"No transfers match your filter selection. [Clear filters](url)"

Reply:
I still think it'd be really useful to show the date range being applied, because at first glance it'll appear immediately broken if there's no transactions and the user knows their finance app had transactions previously.

This is especially true on mobile, where the filters are hidden by default.

Another problem is that we'd need a definition for what "clear filters" mean for the date range. Should we just set it to Nov of last year (since it's when it launched)? Should we be smarter and set it based on the app's initialization block time (actions are only possible on the app after it's been initialized)?

Reply:
Oh right, I was assuming the date range was not applied until the user selects one!

@sohkai I think your solution would be an improvement in the current state of things, but it would still be quite confusing on mobile with the current design in some cases: if there is let’s say one item, and the others are hidden, nothing would indicate the current date range as it is only displayed in the empty state.

About the “clear filters”: what about the case where a specific date range has been selected, but also another filter? Do we reset the date range as well? Or do we consider it not being a filter?

I think it could be easier for the time being to only apply the date range once the user selects it (having the field empty by default), like the other filters, at least until this part gets redesigned.

Thoughts @dizzypaty @sohkai?

Reply:
> I think it could be easier for the time being to only apply the date range once the user selects it (having the field empty by default), like the other filters, at least until this part gets redesigned.

I think this would be preferable, but does the component allow us to do this? It might look weird being completely blank as its default state.

Reply:
@sohkai, @bpierre I also assumed none of filters were applied until users selection. 

Having a look at this with more detail now: 
- Date range should be deselected by default (until user makes a date selection)
- We should indicate today’s date with a slightly different visual style than the date selection – will do a quick mock-up!
- We should include a ‘Clear all’ button, that removes the current filter selection

Reply:
@sohkai , @bpierre Based on the previous comments, these are some changes to the date range filter:

### Default state (no filters applied, calendar displays today's date)
![WebApp-1366px - Finance main view](https://user-images.githubusercontent.com/7205369/55944511-0e12ec80-5c49-11e9-8c86-b0ddb2d5248b.png)
### Date range selected 
![WebApp-1366px - Finance main view-1](https://user-images.githubusercontent.com/7205369/55944563-22ef8000-5c49-11e9-9fe2-7af4c4b9c482.png)
### Filters applied 
![WebApp-1366px - Finance main view-2](https://user-images.githubusercontent.com/7205369/55944580-300c6f00-5c49-11e9-9af7-fc637275cd6e.png)

**Note:** after trying a couple of things yesterday, I think we can leave the 'Clear filters' button out of these changes for the release and then rethink the layout properly. 

[Figma file](https://www.figma.com/file/JUMdeJefkK1TZtYmU8POnSd5/Finance-filters?node-id=0%3A1)